tensorflow/python/ops/nn_ops.py

```diff
@@ -1460,51 +1460,65 @@ def _swap_axis(logits, dim_index, last_index, name=None):
   return output
 
 
-def softmax(logits, dim=-1, name=None):
+def softmax(logits, axis=None, name=None, dim=None):
```

Comment:
You should use the @deprecated_args decorator to mark `dim` deprecated. I am in the process of making that decorator behave less noisy (warn only the first time), but it's already a good idea to use, so usage can adapt to the new names.